Fixed some bugs + merged with dr1s latest changes [commit ff2c7b0]

yosh778 opened the pull request.

yosh778 committed:
Fixes: Fixed some bugs

- Fixed defense rate divide 0 exception
- Fixed encoding for other languages than english
- Fixed auto_build mode
- Fixed main.py error

yosh778 committed:
Fixed last 0 exception after test fail

yosh778 committed:
Merge remote-tracking branch 'dr1s/master'

yosh778 committed:
Fixed minor bug in get_next_building_to_build_on_planet

yosh778 committed:
Restored Twilio SMS messaging

yosh778 committed:
Fixed SMS Activation config parsing

dr1s commented:
FYI this seems to include my scheduler implementation. If this gets merged please check the code again as I'm not sure how well it worked.

yosh778 commented:
Ok indeed, I saw many nice improvements on your master branch so I merged the latest with this master branch + some fixes I did.
The scheduler feature isn't enabled by default (just need to add a section to user.cfg), so it shouldn't be a big issue. I only tested the program in regular mode myself.
Thanks for the nice work btw ! Pretty cool

yosh778 committed:
Fixed SMS indentation

yosh778 committed:
Some language encoding fixes

yosh778 committed:
Some more fixes

yosh778 committed:
Added more exception tolerance to automated tasks

yosh778 committed:
Research: Improved auto_research

Auto-research now attempting research on each planet until
one is started or one is already going on

yosh778 committed:
Improved get_cargo_fleet_for_mission fleet computation

yosh778 committed:
debug

yosh778 committed:
debug

yosh778 committed:
Fixed minor bug

yosh778 committed:
Minor update

yosh778 committed:
Improved attacking planet selection

yosh778 committed:
Fixed builder bug !

yosh778 committed:
Added more exception tolerance to spy reports parsing

yosh778 committed:
Now aborting auto-spy when no more fleet slots are available

yosh778 committed:
Fixed bug in nearest planet to coordinates function, improved it and
added a function to sort planets by nearest to a target
Now attacks will be performed from nearest planets to targets instead of random

yosh778 committed:
Improved spy bot by spying from all nearest planets

yosh778 committed:
Added config file parameter support

yosh778 committed:
Added support for multiple cookies / accounts

yosh778 committed:
Added authentication loss detection and cancelling attack when no more ships

yosh778 committed:
Now only attacking planets with enough resources

yosh778 committed:
Improved some logs

yosh778 committed:
Implemented fleet auto-escape with max resources on incoming attacks

yosh778 committed:
Started code for even better origin planet selection (order by loot, then proximity)

yosh778 committed:
Fixed fleet send bug

yosh778 committed:
Added mission type to fleet movements and now only escaping hostile attack fleets

yosh778 committed:
Now updating slots counters between each attack

yosh778 committed:
Now ordering origin attack planets by predicted loot then proximity for same loots

yosh778 committed:
Fixed latest feature

yosh778 committed:
Switched back to attacking from nearest planets

yosh778 committed:
Improved exception logs

yosh778 committed:
Fixed exception handling issues

yosh778 committed:
Logging fix

yosh778 committed:
Do not auto-build structures if 150 fields already taken on a planet

yosh778 committed:
Some improvements

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fixed some bugs

yosh778 committed:
Some more improvements

yosh778 committed:
Optimization

yosh778 committed:
Fix

yosh778 committed:
Allow moon fleet destruction

yosh778 committed:
+

yosh778 committed:
Fixed general target galaxy parsing (including player rank informations)

yosh778 committed:
Escape fleet fix

yosh778 committed:
+

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Update

yosh778 committed:
debug

yosh778 committed:
Fix

yosh778 committed:
Cleanup

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
Added moons support

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
Fix

yosh778 committed:
+

yosh778 committed:
Added Ogame API server config

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
Moon optim

yosh778 committed:
Moon escape fix

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
Added auto repair / collect ships

yosh778 committed:
No more recyclers to attack : too slow

yosh778 committed:
No more recyclers to attack : too slow

yosh778 committed:
Added day item autobuy

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
Fix

yosh778 committed:
Added player rank to logs

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
+

yosh778 committed:
Optims

yosh778 committed:
Fix

yosh778 committed:
Update README.md

yosh778 committed:
Update README.md

sim2511 commented on ogbot/core/attacker.py:
```diff
@@ -63,50 +60,128 @@ def attack_inactive_planets_from_spy_reports(self, reports):
+            planet.loot = self.fleet_client.get_attack_loot(planet, target)
+
+        ordered_planets = sorted(self.planets, key=lambda x: x.loot, reverse=True)
+        maxLoot = max(self.planets, key=lambda x: x.loot)
```
Thanks for this awesome work to you and Yoshi :)

leogouttefarde commented:
Ok so I made a lot of updates & improvements on my fork, stopped playing recently though cause it's not much fun anymore once you reach a certain level lol

yosh778 commented:
Ok so I made a lot of updates & improvements on my fork, stopped playing recently though cause it's not much fun anymore once you reach a certain level lol

yosh778 committed:
Added some frontend html interface

golyalpha commented on ogbot/core/defender.py:
```diff
@@ -15,7 +16,12 @@ def auto_build_defenses(self):
+            try:
+                self.auto_build_defenses_to_planet(planet)
+
+            except Exception as e:
```
I'd recommend removing the "as e" part of this if you aren't accessing the exception in the except codeblock.

yosh778 replied:
yeah I tried doing that at some point, seems there's still some left though
